lib/logger.js

```diff
@@ -6,449 +8,449 @@
-    }
-
-    if (this._err) {
-        return this._err;
-    }
+      }
 
-    this._bufferLog(message, callback);
-};
-
-Logger.prototype._bufferLog = function(message, callback) {
-    if (!message || !message.line) {
-        debug('Ignoring empty message');
-        return;
-    }
-
-    if (!callback || typeof callback !== 'function') {
-        callback = (err) => { debug(err); };
-    }
+      if (opts.logSourceCRN) {
+        message.logSourceCRN = opts.logSourceCRN
+      }
 
-    if (this._max_length && message.line.length > configs.MAX_LINE_LENGTH) {
-        message.line = message.line.substring(0, configs.MAX_LINE_LENGTH) + ' (cut off, too long...)';
-        debug('Line was longer than ' + configs.MAX_LINE_LENGTH + ' chars and was truncated.');
-    }
+      if (opts.saveServiceCopy) {
+        message.saveServiceCopy = opts.saveServiceCopy
+      }
 
-    this._bufByteLength += sizeof(message);
+      if (opts.appOverride) {
+        message.appOverride = opts.appOverride
+      }
 
-    if (this._bufByteLength >= this._failedBufRetentionLimit) {
-        debug(`buffer size exceeded the failed buffer retention limit. Dropping this line: ${message.line}`);
-        this._bufByteLength -= sizeof(message);
-    } else {
-        debug('Buffering message: %s', message.line);
-        this._buf.push(message);
+      if (this._shimProperties) {
+        this._shimProperties.forEach((prop) => {
+          if (opts.hasOwnProperty(prop)) {
+            message[prop] = opts[prop]
+          }
+        })
+      }
     }
+  } else if (!this._index_meta) {
+    message.meta = stringify(message.meta)
+  }
 
-    if (!this._isLoggingBackedOff && (this._bufByteLength >= this._flushLimit)) {
-        debug('Buffer size meets (or exceeds) flush limit.  Immediately flushing');
-        this._flush((err) => {
-            if (err) {
-                debug('Received an error while flushing...' + err);
-            }
-        });
-    }
+  if (this._err) {
+    return callback(this._err)
+  }
 
-    if (this._isLoggingBackedOff) {
-        debug('Backing off.');
-        this._isLoggingBackedOff = false;
-        this._flusher = setTimeout(() => {
-            this._flush(callback);
-        }, this._retryTimeout);
-    }
+  this._bufferLog(message, callback)
+}
 
-    if (!this._flusher) {
-        debug('No scheduled flush. Scheduling for %d ms from now.', configs.FLUSH_INTERVAL);
-        this._flusher = setTimeout(() => {
-            this._flush(callback);
-        }, this._flushInterval);
-    }
-};
+Logger.prototype._bufferLog = function(message, callback) {
+  if (!callback || typeof callback !== 'function') {
+    callback = (err) => { debug(err) }
+  }
+
+  if (!message || !message.line) {
+    debug('Ignoring empty message')
+    return setImmediate(callback)
+  }
+
+  if (this._max_length && message.line.length > configs.MAX_LINE_LENGTH) {
+    message.line = message.line.substring(0, configs.MAX_LINE_LENGTH) + ' (cut off, too long...)'
```

Comment:
We need to verify with some other folks that it's safe to remove `MAX_LINE_LENGTH` assertion since `IBM` was asking for that too - I removed this limitation from most of open source tools except this and some other code libraries. I think we already do truncation on server-side and there is no need for that here.

Reply:
For sure.  That's going to be part of the next `major` as well, if we can just figure out what's supposed to happen :]